src/Dialogs/StripeDialog.vala

```diff
@@ -322,7 +322,7 @@ public class AppCenter.Widgets.StripeDialog : Gtk.Dialog {
         try {
             switch (entry) {
                 case 0:
-                    var regex = new Regex ("""[a-z|0-9]+@[a-z|0-9]+\.[a-z]+""");
+                    var regex = new Regex ("""([0-9a-zA-Z]([-\.\w]*[0-9a-zA-Z])*@([0-9a-zA-Z][-\w]*[0-9a-zA-Z]\.)+[a-zA-Z]{2,9})""");
```

Comment:
Since we're modifying the regex, we should add the `+` to the regex since at least for Google's emails, you can have `foo+something@gmail.com`

```suggestion
                    var regex = new Regex ("""([0-9a-zA-Z]([-\.\+\w]*[0-9a-zA-Z])*@([0-9a-zA-Z][-\w]*[0-9a-zA-Z]\.)+[a-zA-Z]{2,9})""");
```

Reply:
Right, forgot about that one.